File: SQL Server 2012/Exercises/Chapter 23/C#/ApClrObjectsEx/ApClrObjectsEx/GetTop10Vendors.cs
//------------------------------------------------------------------------------
// <copyright file="CSSqlStoredProcedure.cs" company="Microsoft">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//------------------------------------------------------------------------------
using System;
using System.Data;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;

public partial class StoredProcedures
{
    [Microsoft.SqlServer.Server.SqlProcedure]
    public static void GetTop10Vendors ()
    {
        SqlConnection connection =
            new SqlConnection("Context connection=true");
        connection.Open();

        string selectStatement =
            "SELECT TOP 10 VendorName, SUM(InvoiceTotal) " +
            "AS InvoiceTotal " +
            "FROM Invoices JOIN Vendors " +
            "ON Invoices.VendorID = Vendors.VendorID " +
            "GROUP BY VendorName " +
            "ORDER BY SUM(InvoiceTotal) DESC";

        SqlCommand selectCommand =
            new SqlCommand(selectStatement, connection);

        SqlPipe pipe = SqlContext.Pipe;
        pipe.ExecuteAndSend(selectCommand);

        connection.Close();
    }
}

File: SQL Server 2012/Exercises/Chapter 22/Visual Basic/ApClrObjectsEx/GetTop10Vendors.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server


Partial Public Class StoredProcedures
    <Microsoft.SqlServer.Server.SqlProcedure()> _
    Public Shared Sub  GetTop10Vendors ()
        Dim connection As New SqlConnection("Context connection=true")
        connection.Open()


        Dim sql As String =
            "SELECT TOP 10 VendorName, SUM(InvoiceTotal) " &
            "AS InvoiceTotal " &
            "FROM Invoices JOIN Vendors " &
            "ON Invoices.VendorID = Vendors.VendorID " &
            "GROUP BY VendorName " &
            "ORDER BY SUM(InvoiceTotal) DESC"

        Dim selectCommand As New SqlCommand(sql, connection)

        Dim pipe As SqlPipe = SqlContext.Pipe
        pipe.ExecuteAndSend(selectCommand)

        connection.Close()
    End Sub
End Class
